src/Schema/SchemaInterface.php

```diff
@@ -0,0 +1,489 @@
+
+    /**
+     * Returns all schema names in the database, except system schemas.
+     *
+     * @param bool $refresh whether to fetch the latest available schema names. If this is false, schema names fetched
+     * previously (if available) will be returned.
+     *
+     * @throws NotSupportedException
+     *
+     * @return string[] all schema names in the database, except system schemas.
+     */
+    public function getSchemaNames(bool $refresh = false): array;
+
+    /**
+     * Returns all table names in the database.
+     *
+     * @param string $schema the schema of the tables. Defaults to empty string, meaning the current or default schema
+     * name.
+     * If not empty, the returned table names will be prefixed with the schema name.
+     * @param bool $refresh whether to fetch the latest available table names. If this is false, table names fetched
+     * previously (if available) will be returned.
+     *
+     * @throws NotSupportedException
+     *
+     * @return string[] all table names in the database.
+     */
+    public function getTableNames(string $schema = '', bool $refresh = false): array;
+
+    /**
+     * Determines the PDO type for the given PHP data value.
+     *
+     * @param mixed $data the data whose PDO type is to be determined
+     *
+     * @return int the PDO type
+     *
+     * {@see http://www.php.net/manual/en/pdo.constants.php}
+     */
+    public function getPdoType($data): int;
+
+    /**
+     * Refreshes the schema.
+     *
+     * This method cleans up all cached table schemas so that they can be re-created later to reflect the database
+     * schema change.
+     */
+    public function refresh(): void;
+
+    /**
+     * Refreshes the particular table schema.
+     *
+     * This method cleans up cached table schema so that it can be re-created later to reflect the database schema
+     * change.
+     *
+     * @param string $name table name.
+     */
+    public function refreshTableSchema(string $name): void;
+
+    /**
+     * Returns the ID of the last inserted row or sequence value.
+     *
+     * @param string $sequenceName name of the sequence object (required by some DBMS)
+     *
+     * @throws InvalidCallException if the DB connection is not active
+     *
+     * @return string the row ID of the last row inserted, or the last value retrieved from the sequence object
+     *
+     * @see http://www.php.net/manual/en/function.PDO-lastInsertId.php
+     */
+    public function getLastInsertID(string $sequenceName = ''): string;
+
+    /**
+     * @return bool whether this DBMS supports [savepoint](http://en.wikipedia.org/wiki/Savepoint).
+     */
+    public function supportsSavepoint(): bool;
+
+    /**
+     * Creates a new savepoint.
+     *
+     * @param string $name the savepoint name
+     *
+     * @throws Exception|InvalidConfigException|Throwable
+     */
+    public function createSavepoint(string $name): void;
```

Comment:
Not sure it belongs to schema. That's active action and schema is about getting DDL.

Reply:
it's be fixed in next PR